src/enlyze/client.py

```diff
@@ -265,3 +278,71 @@ def get_timeseries_with_resampling(
             end=end,
             variables=variables_sequence,
         )
+
+    def _get_production_runs(
+        self,
+        *,
+        production_order: Optional[str] = None,
+        product: Optional[str] = None,
+        appliance: Optional[UUID] = None,
+        start: Optional[datetime] = None,
+        end: Optional[datetime] = None,
+    ) -> Iterator[ProductionRun]:
+        """Get production runs from the API."""
+
+        filters = {
+            "production_order": production_order,
+            "product": product,
+            "appliance": appliance,
+            "start": start.isoformat() if start else None,
+            "end": end.isoformat() if end else None,
+        }
+        params = {k: v for k, v in filters.items() if v is not None}
+        return self._production_runs_api_client.get_paginated(
+            "production-runs", ProductionRun, params=params
+        )
+
+    def get_production_runs(
+        self,
+        *,
+        production_order: Optional[str] = None,
+        product: Optional[user_models.Product] = None,
+        appliance: Optional[user_models.Appliance] = None,
+        start: Optional[datetime] = None,
+        end: Optional[datetime] = None,
+    ) -> user_models.ProductionRuns:
+        """Retrieve all production runs.
+
+        :param appliance: The appliance for which to get all production runs.
+        :type appliance: :class:`~enlyze.models.Appliance`
```

Comment:
Why do you explicitly specify the types here? They should be inferred automatically from the arguments.

Reply:
Because we defined it above as well

Reply:
Alright. But in any case, the explicitly defined type here is wrong (`product` and `appliance` are optional). Either fix it or just remove the `:type` altogether. Looks like it would get properly rendered:
![image](https://github.com/enlyze/enlyze-python/assets/224128/d27f9994-5f8d-401f-b876-ef1c24713353)

Reply:
4f34ae5